Fix/improve the example in the project's README

hartzell opened the pull request:
The example in the README didn't work (out of date?).

This:

- fixes the configuration of the ButtonHandler, using a set of ButtonInput objects for the callbacks; and
- makes the example slightly more complex/useful by demonstrating how to use the ButtonInitConfig objects and how to configure a callback for a triple press.

hartzell committed:
Fix/improve the example in the project's README

The example in the README didn't work (out of date?).

This:

- fixes the configuration of the ButtonHandler, using a set of
  ButtonInput objects for the callbacks; and
- makes the example slightly more complex/useful by demonstrating how
  to use the ButtonInitConfig objects and how to configure a callback
  for a triple press.

EGJ-Moorington commented:
Hello and thank you for taking the time to submit a pull request! It is greatly appreciated.

The example in [`README.rst`](https://github.com/EGJ-Moorington/CircuitPython_Button_Handler/blob/ed7e0326fa88e094a21b1bf33b9a5354be97c690/README.rst) is indeed outdated, and should be replaced by [`examples/button_handler_singlebutton.py`](https://github.com/EGJ-Moorington/CircuitPython_Button_Handler/blob/ed7e0326fa88e094a21b1bf33b9a5354be97c690/examples/button_handler_singlebutton.py).

I will not be at home for a few days, but I believe your more complex example could be added as a separate example in the [`examples`](https://github.com/EGJ-Moorington/CircuitPython_Button_Handler/blob/ed7e0326fa88e094a21b1bf33b9a5354be97c690/examples) folder.

hartzell committed:
Rework for EGJ's comments

- Use `examples/button_handler_singlebutton.py` as the example in the
  README.md.
- Add my "triple press with extra delay" example to the examples dir.

hartzell commented:
No rush, now that I figured out how to use it, I'm good to go.  This is just for future me's.

I've reworked it as you suggested.

hartzell committed:
Format the example the way that ruff wants it...

Interestingly, ruff seems to disagree with isort
(or perhaps it's because I'm running with a local
copy of the library so that I could figure out how
it worked...).

EGJ-Moorington commented:
Thanks for fixing the example so that it passes all tests, I was planning on fixing it myself when I got home. I am not familiar with `isort`, but I believe the issue was because [`ruff` places first-party imports last](https://docs.astral.sh/ruff/faq/#how-does-ruffs-import-sorting-compare-to-isort).

hartzell commented:
[isort](https://pypi.org/project/isort/) is the tool that all the cool kids (ok, the OCD tidy kids) used to put Python files into a standard order before `ruff` came on the scene.

It kind of surprises me that ruff did something different, but....

Could also be operator error somehow.

**Thank you** for sharing the library and merging the change.

EGJ-Moorington commented:
Fixes #37.
